modules/eventadd.php

```diff
@@ -154,12 +154,22 @@
 				else
 					$mailfname = '';
 
-				if ($user['email'])
-					$mailfrom = $user['email'];
-				elseif ($qemail = $LMS->GetQueueEmail($ticket['queue']))
-					$mailfrom = $qemail;
-				else
-					$mailfrom =  $ticket['mailfrom'];
+				$helpdesk_sender_email = ConfigHelper::getConfig('phpui.helpdesk_sender_email');
+				if(!empty($helpdesk_sender_email)) {
+					$mailfrom = $helpdesk_sender_email;
+
+					if($mailfrom == 'queue')
+						$mailfrom = $LMS->GetQueueEmail($ticket['queue']);
+					elseif($mailfrom == 'user')
+						$mailfrom = $user['email'];
+				} else {
```

Comment:
@mikenowak let's look for the moment at pointed code, please.
In $helpdesk_sender_email non-empty scenario we set $mailfrom without its source emptiness checks.
Is it correct and intended behavior?

Reply:
should phpui.helpdesk_sender_email conf. setting determine suggested order of attribute usage only?

Reply:
My intention here was to ensure that the the queue address is used, and completely disregard the order.

Reply:
The code
```php
  $helpdesk_sender_email = ConfigHelper::getConfig('phpui.helpdesk_sender_email');
  if(!empty($helpdesk_sender_email)) {
    $mailfrom = $helpdesk_sender_email;

    if($mailfrom == 'queue')
      $mailfrom = $queue['email'];
    elseif($mailfrom == 'user')
      $mailfrom = $user['email'];
  } else {
    $mailfrom = $user['email'] ? $user['email'] : $queue['email'];
  }
```
could be changed to:
```php
  $helpdesk_sender_email = ConfigHelper::getConfig('phpui.helpdesk_sender_email', 'user,queue');
  $attributes = explode(',');
  $attribute = reset($attributes);
  while ($attribute !== false) {
    if ($attribute == 'user') {
      if ($user['email']) {
        $mailfrom = $user['email'];
        break;
      }
    } elseif ($attribute == 'queue') {
      if ($qemail = $LMS->GetQueueEmail($ticket['queue'])) {
        $mailfrom = $qemail;
        break;
      }
    } else {
      $mailfrom = $helpdesk_sender_email;
      break;
    }
  }
```

Reply:
Look at the above proposal and please answer me if it wouldn't have conflict with your use case?

Reply:
Yes, that would be fine - I don't believe it would create a conflict.

Reply:
done with 154a5a1a0